tests/test_anyof_constraints.py

```diff
@@ -0,0 +1,153 @@
+"""Tests for anyOf constraints generated by Required(Any(...)) patterns."""
```

Comment:
Appreciate the detailed tests, though I'd like these to follow the existing test conventions rather than adding one off new files here. (I'm assuming these are vibe coded given the output since it matches my vibe coding results with arbitrary test files, inline imports, and verbose comments.  My suggestion would be to ask the model to review the existing test structure first)